pdcupdater/handlers/retirement.py

```diff
@@ -0,0 +1,171 @@
+        # This query guarantees a unique component branch, so a count of 1 is
+        # expected
+        branch_query_rv = pdc['component-branches']._(
+            name=branch, type=component_type, global_component=repo)
+
+        if branch_query_rv['count'] != 1:
+            log.error('"{0}/{1}" was not found in PDC'.format(namespace, repo))
+            return
+
+        branch = branch_query_rv['results'][0]
+        # If the branch is already EOL in PDC, don't do anything
+        if branch['active'] is False:
+            return
+
+        self._retire_branch(pdc, branch)
+
+    @staticmethod
+    def _retire_branch(pdc, branch):
+        """ Internal method for retiring a branch in PDC. """
+        today = datetime.utcnow().date()
+        for sla in branch['slas']:
+            sla_eol = datetime.strptime(sla['eol'], '%Y-%m-%d').date()
+            if sla_eol > today:
+                pdc['component-branch-slas'][sla['id']]._ \
+                    += {'eol': str(today)}
+
+    @staticmethod
+    def _pdc_to_namespace(pdc_type):
+        pdc_to_namespace = {
+            'rpm': 'rpms',
+            'module': 'modules',
+            'container': 'container',
+        }
+        if pdc_type not in pdc_to_namespace:
+            raise ValueError('The PDC type "{0}" is not supported'
+                             .format(pdc_type))
+        else:
+            return pdc_to_namespace[pdc_type]
+
+    @staticmethod
+    def _is_retired_in_cgit(namespace, repo, branch, requests_session=None):
+        if requests_session is None:
+            requests_session = requests.Session()
+
+        cgit_url = 'https://src.fedoraproject.org/cgit'
+        # Check to see if they have a dead.package file in dist-git
+        url = '{base}/{namespace}/{repo}.git/plain/dead.package?h={branch}'
+        response = requests_session.head(url.format(
+            base=cgit_url,
+            namespace=namespace,
+            repo=repo,
+            branch=branch,
+        ))
+
+        # If there is a dead.package, then the branch is retired in cgit
+        if response.status_code in [200, 404]:
+            return response.status_code == 200
+        else:
+            raise ValueError(
+                'The connection to cgit failed. Retirement status could not '
+                'be determined. The status code was: {0}. The content was: '
+                '{1}'.format(response.status_code, response.content))
+
+    def audit(self, pdc):
+        """ Returns the difference in retirement status in PDC and dist-git.
+
+        This function compares the status in PDC and the status in the
+        "real world" (i.e., in dist-git) and return the difference.
+        """
+        consistent = set()
+        inconsistent = set()
+        session = requests.Session()
+
+        # Look up all non-retired branches from PDC
+        log.info('Looking up all branches from PDC.')
+        branches = pdc.get_paged(pdc['component-branches'])
```

Comment:
The comment says "all non-retired branches" but this query will return *all* branches.